src/Database/Statement.php

```diff
@@ -0,0 +1,370 @@
+<?php
+declare(strict_types=1);
+
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         5.0.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Database;
+
+use Cake\Database\Log\LoggedQuery;
+use InvalidArgumentException;
+use PDO;
+use PDOException;
+use PDOStatement;
+use Psr\Log\LoggerInterface;
+
+class Statement
```

Comment:
Without the nested design, there is no need for an interface. We expect all instances to extend `Statement`.

Reply:
Hmm.. okay. `DriverInterface` is quite redundant too, no one's going to implement their own implementation of it. `Driver` class will be always used.

Reply:
If we have no scenario where we expect users to swap in their own statements then we could do without the interfaces. It does force us to use inheritance in our other statements instead of decoration as well.

Reply:
I updated the design to support inheritance in driver-specific statements better. Notice how `SqlserverStatement` doesn't have to duplicate the bind logic anymore.

Reply:
Re-introduced StatementInterface.